test/e2e/httpproxy/httpproxy_test.go

```diff
@@ -484,4 +484,48 @@ descriptors:
 
 		f.NamespacedTest("grpc-web", testGRPCWeb)
 	})
+
+	Context("global external auth", func() {
+		withGlobalExtAuth := func(body e2e.NamespacedTestBody) e2e.NamespacedTestBody {
+			return func(namespace string) {
+				Context("with global external auth service", func() {
+					BeforeEach(func() {
+						contourConfig.GlobalExternalAuthorization = config.GlobalExternalAuthorization{
+							ExtensionService: fmt.Sprintf("%s/%s", namespace, "testserver"),
+							FailOpen:         false,
+							AuthPolicy: &config.GlobalAuthorizationPolicy{
+								Context: map[string]string{
+									"location": "global_config",
+									"header_2": "message_2",
+								},
+							},
+							ResponseTimeout: "10s",
+						}
+						contourConfiguration.Spec.GlobalExternalAuthorization = &contour_api_v1alpha1.GlobalExternalAuthorizationConfig{
+							ExtensionService: fmt.Sprintf("%s/%s", namespace, "testserver"),
+							FailOpen:         ref.To(false),
+							AuthPolicy: &contour_api_v1alpha1.GlobalAuthorizationPolicy{
+								Disabled: ref.To(false),
+								Context: map[string]string{
+									"location": "global_config",
+									"header_2": "message_2",
+								},
+							},
+							ResponseTimeout: "10s",
+						}
+						require.NoError(f.T(),
+							f.Deployment.EnsureGlobalExternalAuthResources(namespace))
+					})
+					body(namespace)
+				})
+			}
+		}
+
+		f.NamespacedTest("httpproxy-global-ext-auth-non-tls", withGlobalExtAuth(testGlobalExternalAuthVirtualHostNonTLS))
+
+		f.NamespacedTest("httpproxy-global-ext-auth-tls", withGlobalExtAuth(testGlobalExternalAuthTLS))
+
```

Comment:
It is not possible to disable external auth on HTTP Vhosts if GlobalExtAuth is set. I can add a negative test case though

Reply:
I think this is because of this line: https://github.com/projectcontour/contour/blob/main/apis/projectcontour/v1/helpers.go#L23 which is being used by https://github.com/projectcontour/contour/blob/main/apis/projectcontour/v1/helpers.go#L31

If you drop the `v.TLS != nil` check, then it looks like disabling auth for non-TLS vhosts works as expected.

Reply:
I did not notice that check. Yes removing that fixed it. Thanks